pkg/clustermesh/endpointslicesync/endpointslice_test.go

```diff
@@ -262,22 +238,21 @@ func Test_meshEndpointSlice_Reconcile(t *testing.T) {
 		svcStore.CacheStore().Add(svc1)
 		serviceInformer.refreshAllCluster(svc1)
 
-		queue := getControllerQueue(controller)
-		require.NoError(t, waitEmptyQueue(queue))
-
-		// Make sure that we have 1 endpointslice
-		epList, err := getEndpointSlice(&fakeClient, svcName)
-		require.NoError(t, err)
-		require.Equal(t, 1, len(epList.Items))
-
-		svc1.Annotations[annotation.GlobalServiceSyncEndpointSlices] = "false"
-		svcStore.CacheStore().Update(svc1)
-		serviceInformer.refreshAllCluster(svc1)
-
-		require.NoError(t, waitEmptyQueue(queue))
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			// Make sure that we have 1 endpointslice
+			epList, err := getEndpointSlice(&fakeClient, svcName)
+			assert.NoError(c, err)
+			assert.Equal(c, 1, len(epList.Items))
+			svc1.Annotations[annotation.GlobalServiceSyncEndpointSlices] = "false"
+			svcStore.CacheStore().Update(svc1)
+			serviceInformer.refreshAllCluster(svc1)
+
+			assert.EventuallyWithT(t, func(c *assert.CollectT) {
+				epList, err := getEndpointSlice(&fakeClient, svcName)
+				assert.NoError(c, err)
+				assert.Zero(c, len(epList.Items))
+			}, timeout, tick, "endpointslice is not reconciled correctly")
```

Comment:
I'd move this out of the first EventuallyWithT, to simplify troubleshooting possible failures.